trainer/finetune.py

```diff
@@ -0,0 +1,240 @@
+      print(os.path.join(RD_JSONL_DIR, fname))
+      num_rd_examples[split] = preprocessing.rd_jsonl_to_tsv(
+          os.path.join(RD_JSONL_DIR, fname), rd_tsv_path[split])
+    json.dump(num_rd_examples, tf.io.gfile.GFile(rd_counts_path, "w"))
+
+  t5.data.TaskRegistry.add(
+      "rd_recommendations",
+      # Supply a function which returns a tf.data.Dataset.
+      dataset_fn=preprocessing.rd_dataset_fn,
+      splits=["train", "validation"],
+      # Supply a function which preprocesses text from the tf.data.Dataset.
+      text_preprocessor=[preprocessing.conversation_preprocessor],
+      # Use the same vocabulary that we used for pre-training.
+      sentencepiece_model_path=t5.data.DEFAULT_SPM_PATH,
+      # Lowercase targets before computing metrics.
+      postprocess_fn=t5.data.postprocessors.lower_text, 
+      # We'll use accuracy as our evaluation metric.
+      metric_fns=[t5.evaluation.metrics.accuracy, t5.evaluation.metrics.bleu],
+      # Not required, but helps for mixing and auto-caching.
+      num_input_examples=num_rd_examples)
+
+  # Set parallelism and batch size to fit on v2-8 TPU (if possible).
+  # Limit number of checkpoints to fit within 5GB (if possible).
+  model_parallelism, train_batch_size, keep_checkpoint_max = {
+      "small": (1, 256, 16),
+      "base": (2, 128, 8),
+      "large": (8, 64, 4),
+      "3B": (8, 16, 1),
+      "11B": (8, 16, 1)}[FLAGS.size]
+
+  tf.io.gfile.makedirs(MODEL_DIR)
+  # The models from the t5 paper paper are based on the Mesh Tensorflow Transformer.
+  model = t5.models.MtfModel(
+      model_dir=MODEL_DIR,
+      tpu=TPU_ADDRESS,
+      tpu_topology=TPU_TOPOLOGY,
+      model_parallelism=model_parallelism,
+      batch_size=train_batch_size,
+      sequence_length={"inputs": INPUT_LENGTH, "targets": TARGET_LENGTH},
+      learning_rate_schedule=0.003,
+      save_checkpoints_steps=5000,
+      keep_checkpoint_max=keep_checkpoint_max,
+      iterations_per_loop=100,
+  )
+
+  FINETUNE_STEPS = FLAGS.steps #@param {type: "integer"}
+
+  if FLAGS.mode == "all" or FLAGS.mode == "train":
+    model.finetune(
+        mixture_or_task_name="rd_recommendations",
+        pretrained_model_dir=PRETRAINED_DIR,
+        finetune_steps=FINETUNE_STEPS
+    )
+  tf.compat.v1.disable_eager_execution()
+
+  # Evaluate and save predictions
+  if FLAGS.mode == "all" or FLAGS.mode == "evaluate":
+    model.batch_size = train_batch_size * 4 # a larger batch size requires less memory.
+    model.eval(
+        mixture_or_task_name="rd_recommendations",
+        checkpoint_steps="all"
+    )
+
+    save_metrics("rd_recommendations")
+
+
+  # Export the SavedModel
+  export_dir = os.path.join(MODEL_DIR, "export")
+
+  model.batch_size = 1 # make one prediction per call
+  saved_model_path = model.export(
+      export_dir,
+      checkpoint_step=-1,  # use most recent
+      beam_size=1,  # no beam search
+      temperature=1.0,  # sample according to predicted distribution
```

Comment:
Seems useful to be made a flag.

Reply:
Done. Added flags beam_size (default 1) and temperature default (1.0)